browser/resources/brave_new_tab_page_refresh/api/background_impl.ts

```diff
@@ -25,10 +25,8 @@ export function initializeBackgrounds(
         loadTimeData.getString('sponsoredRichMediaBaseUrl')
   })
 
-  function updateCurrentBackground() {
-    store.update({
-      currentBackground: getCurrentBackground(store.getState())
-    })
+  function resetRandomizer() {
+    store.update({ backgroundRandomizer: Math.random() })
   }
```

Comment:
hey, I can see why you've gone this way, but it definitely feels a bit off in a React context.

Just fwiw, if we were to implement this in native React we'd probably do something like this:

```
const backgroundIndex = useMemo(() => Math.floor(Math.random() * backgrouds.length), [background.type, backgrounds.length])
```

Reply:
Yep - putting the random value in state wasn't my first choice, but I kept running into annoying edge cases (even with `React.memo`), probably because of the impurity of using `Math.random`. Once I eliminated the impurity, it worked much better.

Reply:
Changes to "current background" have been removed from this PR and will be addressed in https://github.com/brave/brave-browser/issues/46034.